src/app/Services/SearchService.php

```diff
@@ -15,7 +15,7 @@ class SearchService
     public function search(string $query): array
     {
         $userResults = User::where('name', 'like', "%$query%")->get();
-        $postResults = Post::where('content', 'like', "%$query%")->get();
+        $postResults = Post::whereRaw("content REGEXP '[[:<:]]" . $query . "[[:>:]]'")->get();
```

Comment:
For future references: Be very careful of request queries. Find time to study SQL injection and how to propagate better query builders.